src/lib/pci_virtio_sock.c

```diff
@@ -1381,6 +1384,7 @@ static void pci_vtsock_proc_tx(struct pci_vtsock_softc *sc,
 			goto do_rst;
 		}
 		vq_relchain(vq, idx, 0);
+		sock->credit_cnt = sock->fwd_cnt - sock->credit_limit;
```

Comment:
A comment indicating that this is the fudge in the commit message would be good, I think.